cylc/flow/network/publisher.py

```diff
@@ -70,7 +69,6 @@ def _socket_options(self):
     def _bespoke_stop(self):
         """Bespoke stop items."""
         LOG.debug('stopping zmq publisher...')
-        self.stopping = True
```

Comment:
Method is now a dud?

Reply:
Still called, and adds that log entry for publisher stop  (the Base class (`ZMQSocketBase`) has the `stop()` method for all the derivatives) .

Reply:
In that case is there any reason to override the `ZMQSocketBase._bespoke_stop()`? I can't see the harm in it setting `self.stopping = True` even if it's not used

Reply:
Fair enough.

Reply:
reinstated.

Reply:
Ah I thought there was no point in overriding this method in `WorkflowPublisher` at all because I didn't spot that the log message was slightly different for `ZMQSocketBase`

https://github.com/cylc/cylc-flow/blob/f9efea2e649ffd0ffc1fe2cfc63a122034813412/cylc/flow/network/__init__.py#L293-L296

https://github.com/cylc/cylc-flow/blob/f9efea2e649ffd0ffc1fe2cfc63a122034813412/cylc/flow/network/publisher.py#L69-L72

Could just do
```python
class ZMQSocketBase:
    ...
    def _bespoke_stop(self):
        """Bespoke stop items."""
        LOG.debug(f'stopping {type(self).__name__}...')
        self.stopping = True
```
and not have to override. Not a big deal

Reply:
Yeah, the purpose for the override is not the logging (the replier does something different).. So happy as is..